ci: test-e2e flaky failure in TestE2ETestSuite (auth)

## Summary

The `test-e2e` CI job intermittently fails with `timeout exceeded waiting for block` in `TestE2ETestSuite` (auth e2e tests). The test network's validators fail to produce blocks in time, followed by a panic during cleanup when the WAL file is missing.

## Error

```
--- FAIL: TestE2ETestSuite (6.37s)
    suite.go:56:
        Error: Received unexpected error:
               timeout exceeded waiting for block
        Test: TestE2ETestSuite

panic: open /tmp/TestE2ETestSuite.../node0/simd/data/cs.wal: no such file or directory
```

Stack trace points to `celestia-core`'s `libs/autofile/group.go:365` in `readGroupInfo`.

## Impact

This is a flaky test — it passes on most CI runs but occasionally fails. For example, it failed on PR #719 but passed on PRs #717 and #718.

## Potential fixes

1. Increase the block wait timeout in `testutil/network/network.go`
2. Add retry logic for the block wait in e2e test setup
3. Upstream fix in celestia-core to handle missing WAL files gracefully during cleanup

--- a/testutil/network/network.go
+++ b/testutil/network/network.go
@@ -668,7 +668,7 @@ func (n *Network) LatestHeight() (int64, error) {
 	ticker := time.NewTicker(time.Second)
 	defer ticker.Stop()
 
-	timeout := time.NewTimer(time.Second * 5)
+	timeout := time.NewTimer(time.Second * 30)
 	defer timeout.Stop()
 
 	var latestHeight int64

